dataprep-tools/filecoin/boost_create_deals.py

```diff
@@ -81,6 +85,90 @@
 online = environ.get("DEALTYPE") == "online"
 
 
+def get_chain_head():
+    response = requests.post(
+        url=rpc_endpoint,
+        json={"jsonrpc": "2.0", "id": 1, "method": "Filecoin.ChainHead", "params": []},
+    )
+    assert response.status_code == 200, f"failed to get chain head: {response.text}"
+
+    resp = response.json()
+    assert "result" in resp, f"result key not found in ChainHead response: {resp}"
+    assert (
+        "Height" in resp["result"]
+    ), f"Height key not found in ChainHead response: {resp}"
+
+    return resp["result"]["Height"]
+
+
+def get_collateral(padded_size, verified=True):
+    response = requests.post(
+        url=rpc_endpoint,
+        json={
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "Filecoin.StateDealProviderCollateralBounds",
+            "params": [padded_size, verified, []],
+        },
+    )
+    assert response.status_code == 200, f"failed to get chain head: {response.text}"
+
+    resp = response.json()
+    assert (
+        "result" in resp
+    ), f"result key not found in StateDealProviderCollateralBounds response: {resp}"
+    assert (
+        "Min" in resp["result"]
+    ), f"Min key not found in StateDealProviderCollateralBounds response result: {resp}"
+
+    return resp["result"]["Min"] * 1.2  # add 20% to the min
+
+
+def get_providers(
+    piece_cid,
+    size,
+    padded_size,
+):
+
+    head = get_chain_head()
+    start_epoch = head + start_epoch_head_offset
+
+    provider_collateral = get_collateral(padded_size=padded_size, verified=verified)
+
+    api_key = environ.get("CID_GRAVITY_KEY")
+    headers = {"X-API=KEY": api_key}
+
+    response = requests.post(
+        url="https://service.cidgravity.com/private/v1/get-best-available-providers",
+        headers=headers,
+        json={
+            "pieceCid": piece_cid,
+            "startEpoch": start_epoch,
+            "duration": 1468800,
+            "storagePricePerEpoch": 0,
+            "providerCollateral": provider_collateral,
+            "verifiedDeal": True,
+            "transferSize": size,
+            "transferType": "http",
+            "removeUnsealedCopy": False,
+        },
+    )
+
+    assert (
+        response.status_code == 200
+    ), f"failed to get provider from CID gravity: {response.text}"
+
+    resp = response.json()
+    assert (
+        "result" in resp
+    ), f"result key not found in response from CID gravity: {resp}"
+    assert (
+        "providers" in resp["result"]
+    ), f"providers key not found in response from CID gravity: {resp}"
+
+    return resp["result"]["providers"]
```

Comment:
Might be nice to check if `providers` is empty, and log `reason`.

Reply:
done